File: lv1-library-loan/csharp/LibraryLoan.Application/Books/InMemoryBookRepository.cs
using LibraryLoan.Domain.Books;

namespace LibraryLoan.Application.Books;

public sealed class InMemoryBookRepository : IBookRepository
{
    private readonly Dictionary<BookId, Book> _books = new();

    public Book? Find(BookId id) => _books.GetValueOrDefault(id);

    public void Save(Book book) => _books[book.Id] = book;
}

File: lv1-library-loan/typescript/src/application/books/in-memory-book-repository.ts
import type { Book } from "../../domain/books/book.js";
import type { BookId } from "../../domain/books/book-id.js";
import type { BookRepository } from "./book-repository.js";

export class InMemoryBookRepository implements BookRepository {
  private readonly books = new Map<string, Book>();

  find(id: BookId): Book | undefined {
    return this.books.get(id.value);
  }

  save(book: Book): void {
    this.books.set(book.id.value, book);
  }
}
